app/presenters/two_factor_authentication/set_up_selection_presenter.rb

```diff
@@ -0,0 +1,94 @@
+module TwoFactorAuthentication
+  class SetUpSelectionPresenter
+    include ActionView::Helpers::TranslationHelper
+
+    attr_reader :user
+
+    def initialize(user:)
+      @user = user
+    end
+
+    def render_in(view_context, &block)
+      view_context.capture(&block)
+    end
+
+    def type
+      method.to_s
+    end
+
+    def label
+      case type
+      when 'auth_app'
```

Comment:
If there's an opportunity, I think it would be great if we could get rid of these giant switches and move them into the individual MFA classes? Maybe even an option to flip the proposed refactoring so that we still have one base class, but at least the individual MFA classes are responsible for the logic of switching between label results based on set-up vs. sign-in.